presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/TaskProcessors.java

```diff
@@ -62,59 +62,37 @@ private TaskProcessors() {}
         };
     }
 
-    public static PairFlatMapFunction<Iterator<Tuple2<Integer, PrestoSparkRow>>, Integer, PrestoSparkRow> createTaskProcessor(
+    public static Function<List<Iterator<Tuple2<Integer, PrestoSparkRow>>>, Iterator<Tuple2<Integer, PrestoSparkRow>>> createTaskProcessor(
```

Comment:
Instead of using a general function type. I am wondering if we can have `FlatMapFunctionNSameInputType<Iterator<Tuple2<Integer, PrestoSparkRow>>, Tuple2<Integer, PrestoSparkRow>>`? Essentially thinking about generalizing `FlatMapFunction2` to N inputs with same type :)

Reply:
Could you please elaborate a little more on this?

Reply:
Actually never-mind. I think I got it. Let me have a look

Reply:
Currently the functions argument is `List<Iterator<Tuple2<Integer, PrestoSparkRow>>>`. It's because the inputs are always `Tuple2<Integer, PrestoSparkRow>`. 

When I was thinking about bucketed tables implementation, I envisioned the second type of inputs, `Tuple2<Integer, Split>` -> `(partition,split)`. Thus I would love to leave the flexibility to freely define signature, since it will not necessarily be accepting the inputs of the same type.

Reply:
@arhimondr : Makes sense. We can revisit potential refactors once bucketed table join unbucketed table is implemented.